app/views/auth.py

```diff
@@ -1,13 +1,70 @@
-from flask import Blueprint, redirect, render_template, url_for
-from flask_login import login_required, login_user, logout_user
+from functools import wraps
+
+from flask import Blueprint
+from flask import current_app as app
+from flask import flash, redirect, render_template, url_for
+from flask_login import current_user, login_required, login_user, logout_user
 
 from app import User, db
 from app.forms import LoginForm, RegisterForm
+from app.utils import send_email
 
 auth_bp = Blueprint("auth_bp", __name__)
 
 
+# only allow confirmed users, otherwise redirect to unconfirmeed page
+def confirmed_required(func):
+    @wraps(func)
+    @login_required
+    def inner(*args, **kwargs):
+        if not app.config["ADVANCED_ACCESS_CONTROL"]:
+            return func(*args, **kwargs)
+        if not current_user.confirmed:
+            return redirect(url_for("auth_bp.unconfirmed"))
+        return func(*args, **kwargs)
+
+    return inner
+
+
+# only allow unconfirmed users, otherwise redirect to main page
+def unconfirmed_required(func):
+    @wraps(func)
+    @login_required
+    def inner(*args, **kwargs):
+        if not app.config["ADVANCED_ACCESS_CONTROL"]:
+            return func(*args, **kwargs)
+        if current_user.confirmed:
+            return redirect(url_for("find_page_bp.find_page"))
+        return func(*args, **kwargs)
+
+    return inner
+
+
+# only allow for anonymous (unlogged) users, otherwise redirect to main page
+def anonymous_required(func):
+    @wraps(func)
+    def inner(*args, **kwargs):
+        if not app.config["ADVANCED_ACCESS_CONTROL"]:
+            return func(*args, **kwargs)
+        if not current_user.is_anonymous:
+            return redirect(url_for("find_page_bp.find_page"))
+        return func(*args, **kwargs)
+
+    return inner
+
+
+# THIS IS EXCLUSIVELY FOR TESTING AND MUST BE REMOVED IN THE FINAL VERSION
+@auth_bp.route("/forced_entry", methods=["POST"])
```

Comment:
Endpointy nazywajmy z myślnikiem, a nie podłoga

Reply:
To i tak jest do wyrzucenia ostatecznie ale dobrze szefie